diesel_migrations/migrations_internals/src/lib.rs

```diff
@@ -348,19 +348,13 @@ where
 
     let run_in_transaction = migration
         .metadata()
-        .and_then(|m| m.get("run_in_transaction"));
-    let run_in_transaction = match run_in_transaction.as_ref().map(AsRef::as_ref) {
-        Some("true") => true,
-        Some("false") | None => false,
-        Some(_) => {
-            return Err(MigrationError::InvalidMetadata(
-                "run_in_transaction must be a boolean".into(),
-            )
-            .into())
-        }
-    };
+        .and_then(|m| {
+            m.get("run_in_transaction")
+                .and_then(|x: &dyn Any| x.downcast_ref::<bool>())
```

Comment:
This should return an error if the value wasn't a boolean, not silently ignore it.